contrib/jenkins_tests/vg.sh

```diff
@@ -4,79 +4,120 @@ source $(dirname $0)/globals.sh
+#	test_ip_list="${test_ip_list} ib:$(do_get_ip 'ib' 'mlx5')"
+#fi
+if [ ! -z $(do_get_ip 'eth' 'mlx5') ]; then
+	test_ip_list="${test_ip_list} eth:$(do_get_ip 'eth' 'mlx5')"
+fi
+test_list="tcp:--tcp udp:"
+test_lib=${vg_dir}/install/lib/libvma.so
 test_app=sockperf
+test_app_path=${test_dir}/sockperf/install/bin/sockperf
 
-if [ $(command -v $test_app >/dev/null 2>&1 || echo $?) ]; then
-	test_app=${test_dir}/sockperf/install/bin/sockperf
-	if [ $(command -v $test_app >/dev/null 2>&1 || echo $?) ]; then
-	    echo can not find $test_app
-	    exit 1
+if [ $(command -v $test_app_path >/dev/null 2>&1 || echo $?) ]; then
+	test_app_path=sockperf
+	if [ $(command -v $test_app_path >/dev/null 2>&1 || echo $?) ]; then
+		echo can not find $test_app_path
+		exit 1
 	fi
 fi
 
-vg_args="-v --log-file=${vg_dir}/valgrind.log \
-    --memcheck:leak-check=full --track-origins=yes --read-var-info=yes \
-    --undef-value-errors=yes --db-attach=no --track-fds=yes --show-reachable=yes \
-    --num-callers=32 \
-    --fullpath-after=${WORKSPACE} \
-    --suppressions=${WORKSPACE}/contrib/valgrind/valgrind_vma.supp \
-    --suppressions=${WORKSPACE}/contrib/valgrind/valgrind_libc.supp \
-    --suppressions=${WORKSPACE}/contrib/valgrind/valgrind_sockperf.supp \
-    --suppressions=${WORKSPACE}/contrib/valgrind/valgrind_rdma.supp \
-    "
-
-vg_tests=1
-
-eval "env VMA_TX_BUFS=20000 VMA_RX_BUFS=20000 LD_PRELOAD=$test_lib $test_app sr --tcp -i ${test_ip} > /dev/null 2>&1 &"
-sleep 5
-
-eval "env VMA_TX_BUFS=20000 VMA_RX_BUFS=20000 LD_PRELOAD=$test_lib valgrind $vg_args $test_app pp --tcp -i ${test_ip} -m202 -t5"
-
-pkill -9 sockperf
-
-do_archive "${vg_dir}/valgrind.log"
-
 vg_tap=${WORKSPACE}/${prefix}/vg.tap
+v1=$(echo $test_list | wc -w)
+v1=$(($v1*$(echo $test_ip_list | wc -w)))
+echo "1..$v1" > $vg_tap
+
+nerrors=0
+
+for test_link in $test_ip_list; do
+	for test in $test_list; do
+		IFS=':' read test_n test_opt <<< "$test"
+		IFS=':' read test_in test_ip <<< "$test_link"
+		test_name=${test_in}-${test_n}
+
+		vg_args="-v \
+			--memcheck:leak-check=full --track-origins=yes --read-var-info=yes \
+			--errors-for-leak-kinds=definite --show-leak-kinds=definite,possible \
+			--undef-value-errors=yes --track-fds=yes --num-callers=32 \
+			--fullpath-after=${WORKSPACE} --gen-suppressions=all \
+			--suppressions=${WORKSPACE}/contrib/valgrind/valgrind_vma.supp \
+			"
+		eval "LD_PRELOAD=$test_lib \
+			valgrind --log-file=${vg_dir}/${test_name}-valgrind-sr.log $vg_args \
+			$test_app_path sr ${test_opt} -i ${test_ip} > /dev/null 2>&1 &"
+		sleep 20
+		eval "LD_PRELOAD=$test_lib \
+			valgrind --log-file=${vg_dir}/${test_name}-valgrind-cl.log $vg_args \
+			$test_app_path pp ${test_opt} -i ${test_ip} -t 10"
+
+		if [ `ps -ef | grep $test_app | wc -l` -gt 1 ];
+		then
+			sudo pkill -SIGINT -f $test_app
+			sleep 10
+			# in case SIGINT didn't work
+			if [ `ps -ef | grep $test_app | wc -l` -gt 1 ];
+			then
+				sudo pkill -15 -f $test_app
```

Comment:
@rafiw please use `pkill -15 -f $test_app 2>/dev/null || true` to suppress exit code form pkill